bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/internal/AudioManagerImpl.java

```diff
@@ -0,0 +1,283 @@
+import org.eclipse.smarthome.core.audio.URLAudioStream;
+import org.eclipse.smarthome.core.audio.UnsupportedAudioFormatException;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This service provides functionality around audio services and is the central service to be used directly by others.
+ *
+ * @author Karel Goderis - Initial contribution and API
+ * @author Kai Kreuzer - removed unwanted dependencies
+ */
+public class AudioManagerImpl implements AudioManager, ConfigOptionProvider {
+
+    private static final String SOUND_DIR = "sounds";
+
+    // constants for the configuration properties
+    private static final String CONFIG_URI = "system:audio";
+    private static final String CONFIG_DEFAULT_SINK = "defaultSink";
+    private static final String CONFIG_DEFAULT_SOURCE = "defaultSource";
+
+    private final Logger logger = LoggerFactory.getLogger(AudioManager.class);
+
+    // service maps
+    private Map<String, AudioSource> audioSources = new ConcurrentHashMap<>();
+    private Map<String, AudioSink> audioSinks = new ConcurrentHashMap<>();
+
+    /**
+     * default settings filled through the service configuration
+     */
+    private String defaultSource;
+    private String defaultSink;
+
+    protected void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    protected void deactivate() {
+    }
+
+    protected void modified(Map<String, Object> config) {
+        if (config != null) {
+            this.defaultSource = config.containsKey(CONFIG_DEFAULT_SOURCE)
+                    ? config.get(CONFIG_DEFAULT_SOURCE).toString() : null;
+            this.defaultSink = config.containsKey(CONFIG_DEFAULT_SINK) ? config.get(CONFIG_DEFAULT_SINK).toString()
+                    : null;
+        }
+    }
+
+    @Override
+    public void play(AudioStream audioStream) {
+        play(audioStream, null);
+    }
+
+    @Override
+    public void play(AudioStream audioStream, String sinkId) {
+        if (audioStream != null) {
+            AudioSink sink = getSink(sinkId);
+
+            if (sink != null) {
+                try {
+                    sink.process(audioStream);
+                } catch (UnsupportedAudioFormatException e) {
+                    logger.error("Error playing '{}': {}", audioStream.toString(), e.getMessage());
+                }
+            } else {
+                logger.warn("Failed playing audio stream '{}' as no audio sink was found.", audioStream.toString());
+            }
+        }
+    }
+
+    @Override
+    public void playFile(String fileName) throws AudioException {
+        playFile(fileName, null);
+    }
+
+    @Override
+    public void playFile(String fileName, String sink) throws AudioException {
+        File file = new File(
+                ConfigConstants.getConfigFolder() + File.separator + SOUND_DIR + File.separator + fileName);
```

Comment:
I know it is only moved code, but FYI: Theres a method for such path building: https://docs.oracle.com/javase/7/docs/api/java/nio/file/Paths.html#get(java.lang.String,%20java.lang.String...)